### PR TITLE
forced PIM to build with jdk11

### PR DESCRIPTION
**Pull request by mbiarnes**
**Thank you for submitting this pull request**

in all daily-build PIM will be skipped first but after the build of all other repositories it will be built with jdk11. So it will be also built with jdk11 when the java-tool is defined jdk8.
The jdk11-db-main and jdk11-prod-main are built both with jdk11and PIM is needed here. OK
The jdk8-db-main is built with jdk8 but PIM will be built with jdk11.

<details>
<summary>
How to replicate CI configuration locally?
</summary>

Build Chain tool does "simple" maven build(s), the builds are just Maven commands, but because the repositories relates and depends on each other and any change in API or class method could affect several of those repositories there is a need to use [build-chain tool](https://github.com/kiegroup/github-action-build-chain) to handle cross repository builds and be sure that we always use latest version of the code for each repository.
 
[build-chain tool](https://github.com/kiegroup/github-action-build-chain) is a build tool which can be used on command line locally or in Github Actions workflow(s), in case you need to change multiple repositories and send multiple dependent pull requests related with a change you can easily reproduce the same build by executing it on Github hosted environment or locally in your development environment. See [local execution](https://github.com/kiegroup/github-action-build-chain#local-execution) details to get more information about it.
</details>

<details>
<summary>
How to retest this PR or trigger a specific build:
</summary>

* <b>a pull request</b> please add comment: <b>Jenkins retest</b> (using <i>this</i> e.g. <b>Jenkins retest this</b> optional but no longer required)
 
* for a <b>full downstream build</b> 
  * for <b>jenkins</b> job: please add comment: <b>Jenkins run fdb</b>
  * for <b>github actions</b> job: add the label `run_fdb`
    
* <b>a compile downstream build</b> please  add comment: <b>Jenkins run cdb</b>

* <b>a full production downstream build</b> please add comment: <b>Jenkins execute product fdb</b>

* <b>an upstream build</b> please add comment: <b>Jenkins run upstream</b>
</details>

### CODE REVIEW

**Comment by mareknovotny** [comment]
could you please fix/switch https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/01c1647c3a405c6c386a5d686dc791fb0d7e159e/.ci/jenkins/Jenkinsfile.daily#L135 to not show all files added to zip archive? `-q` is i think the option

**Comment by mareknovotny** [comment]
so do we want to have daily-jdk8 with jdk11 built PIM? i would keep it skipped only, not sure what is the need here as the daily-jdk11 will deploy it too?

File: .ci/jenkins/Jenkinsfile.daily [hunk 1]
```diff
@@ -106,6 +106,17 @@ pipeline {
                 }
             }
         }
+        stage('Deploy process-migration-service locally with jdk11'){
+            steps {
+                dir("${WORKSPACE}" + 'kiegroup/process-migration-service') {
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.daily:111]: I would say process-migration-service folder it's different than the one you are proposing...
```suggestion
                dir("${WORKSPACE}/bc/kiegroup_process_migration_service') {
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
please review

File: .ci/jenkins/Jenkinsfile.daily [hunk 2]
```diff
@@ -106,6 +106,17 @@ pipeline {
                 }
             }
         }
+        stage('Deploy process-migration-service locally with jdk11'){
+            steps {
+                dir("${WORKSPACE}/bc/kiegroup_process_migration_service") {
+                    configFileProvider([configFile(fileId: '771ff52a-a8b4-40e6-9b22-d54c7314aa1e', targetLocation: 'jenkins-settings.xml', variable: 'SETTINGS_XML_FILE')]) {
+                        withEnv(["JAVA_HOME=${tool 'kie-jdk11'}", "PATH=${tool 'kie-jdk11'}/bin:${env.PATH}"]) {
```
Comment (Ginxo) [inline .ci/jenkins/Jenkinsfile.daily:113]: sorry, I missed that part from here, why don't you use this approach https://github.com/kiegroup/droolsjbpm-build-bootstrap/blob/e469889a614b32fc4e94030e4f799248763e1287/.ci/jenkins/Jenkinsfile.buildchain#L79? I think it's more clear, something like
```
stage('Deploy process-migration-service locally with jdk11'){
    tools {
      jdk 'kie-jdk11'
    }
    steps {
        dir("${WORKSPACE}/bc/kiegroup_process_migration_service") {
            configFileProvider([configFile(fileId: '771ff52a-a8b4-40e6-9b22-d54c7314aa1e', targetLocation: 'jenkins-settings.xml', variable: 'SETTINGS_XML_FILE')]) {
                  sh 'mvn -B -e -U clean deploy -Dfull -Drelease -DaltDeploymentRepository=local::default::file://$WORKSPACE/deploy-dir -s $SETTINGS_XML_FILE -Dkie.maven.settings.custom=$SETTINGS_XML_FILE -Dmaven.test.redirectTestOutputToFile=true -Dmaven.test.failure.ignore=true -Dgwt.memory.settings="-Xmx10g" --clean-up-script="$WORKSPACE/clean-up.sh"'
            }
        }
    }
}
```

**Review by Ginxo — CHANGES_REQUESTED** [review]
please check

File: .ci/jenkins/Jenkinsfile.daily [hunk 3]
```diff
@@ -121,7 +132,7 @@ pipeline {
                 withCredentials([usernameColonPassword(credentialsId: 'unpacks-zip-on-qa-nexus', variable: 'kieUnpack')]) {
                     // unpack zip to QA Nexus
                     sh 'cd $WORKSPACE/deploy-dir\n' +
-                       'zip -r kiegroup .\n' +
+                       'zip -qr kiegroup .\n' +
```
Comment (mareknovotny) [inline .ci/jenkins/Jenkinsfile.daily:135]: thanks for that

File: .ci/jenkins/Jenkinsfile.daily (hunk 2 above)
Comment (mareknovotny) [reply .ci/jenkins/Jenkinsfile.daily:113]: but i have still the same question if we need to add this stage and do not let it be without building? no answer yet

**Pull request merged**